miden/tests/integration/stdlib/math/secp256k1_base_field.rs

```diff
@@ -0,0 +1,562 @@
+    ///
+    /// See https://github.com/itzmeanjan/secp256k1/blob/6e5e654823a073add7d62b21ed88e9de9bb06869/field/base_field.py#L78-L96
+    fn neg(self) -> Self::Output {
+        let mut b = [0u32; 8];
+        let mut borrow = 0u32;
+
+        let prime = [
+            4294966319, 4294967294, 4294967295, 4294967295, 4294967295, 4294967295, 4294967295,
+            4294967295,
+        ];
+
+        let v = Self::sbb(prime[0], self.limbs[0], borrow);
+        borrow = v.0;
+        b[0] = v.1;
+
+        let v = Self::sbb(prime[1], self.limbs[1], borrow);
+        borrow = v.0;
+        b[1] = v.1;
+
+        let v = Self::sbb(prime[2], self.limbs[2], borrow);
+        borrow = v.0;
+        b[2] = v.1;
+
+        let v = Self::sbb(prime[3], self.limbs[3], borrow);
+        borrow = v.0;
+        b[3] = v.1;
+
+        let v = Self::sbb(prime[4], self.limbs[4], borrow);
+        borrow = v.0;
+        b[4] = v.1;
+
+        let v = Self::sbb(prime[5], self.limbs[5], borrow);
+        borrow = v.0;
+        b[5] = v.1;
+
+        let v = Self::sbb(prime[6], self.limbs[6], borrow);
+        borrow = v.0;
+        b[6] = v.1;
+
+        let v = Self::sbb(prime[7], self.limbs[7], borrow);
+        b[7] = v.1;
+
+        Self::Output { limbs: b }
+    }
+}
+
+impl Sub for BaseField {
+    type Output = BaseField;
+
+    /// Computes modular subtraction of one secp256k1 base field element, from another one, in Montgomery form
+    ///
+    /// See https://github.com/itzmeanjan/secp256k1/blob/6e5e654823a073add7d62b21ed88e9de9bb06869/field/base_field.py#L98-L102
+    fn sub(self, rhs: Self) -> Self::Output {
+        self + (-rhs)
+    }
+}
+
+impl PartialEq for BaseField {
+    /// Checks whether two secp256k1 field elements are equal or not, in Montogomery form
+    fn eq(&self, other: &Self) -> bool {
+        let mut flg = false;
+
+        for i in 0..8 {
+            flg |= (self.limbs[i] ^ other.limbs[i]) != 0;
+        }
+
+        !flg
+    }
+
+    /// Checks whether two secp256k1 field elements are not equal to each other, in Montogomery form
+    fn ne(&self, other: &Self) -> bool {
+        !(self == other)
+    }
+}
+
+#[test]
+fn test_montgomery_repr() {
+    let source = "
+    use.std::math::secp256k1_base_field
+
+    begin
+        exec.secp256k1_base_field::to_mont
+        exec.secp256k1_base_field::from_mont
+    end";
+
+    let mut num = [0u32; 8];
+    for i in 0..4 {
+        let a = rand_utils::rand_value::<u32>();
+        let b = rand_utils::rand_value::<u32>();
+
+        num[i] = a;
+        num[i ^ 4] = b;
+    }
+
+    let mut stack = [0u64; 8];
+    for i in 0..4 {
+        stack[i] = num[i] as u64;
+        stack[i ^ 4] = num[i ^ 4] as u64;
+    }
+    stack.reverse();
+
+    let test = build_test!(source, &stack);
+    let strace = test.get_last_stack_state();
+
+    for i in 0..8 {
+        assert_eq!(Felt::new(num[i] as u64), strace[i]);
+    }
```

Comment:
This could be simplified in the following ways and I think it would be a bit easier to read & more consistent with other tests in the codebase.
- use `expect_stack` instead of getting the last state and manually comparing.
- only build the stack items once instead of twice.
- I think using `i + 4` instead of `i ^ 4` is more immediately explicit about what's going on with the indexing if anyone new is looking through the tests.
- also a nit: Bobbin has expressed a preference for using underscores before types,e.g. "0_u64" instead of "0u64", so we've been trying to follow this convention elsewhere in the codebase

The simplified test would look like this:

```
    let mut stack = [0_u64; 8];
    for i in 0..4 {
        let a = rand_utils::rand_value::<u32>();
        let b = rand_utils::rand_value::<u32>();

        stack[i] = a;
        stack[i + 4] = b;
    }

    let test = build_test!(source, &stack);
    stack.reverse();
    test.expect_stack(&stack);
```

Reply:
> I think using i + 4 instead of i ^ 4 is more immediately explicit about what's going on with the indexing if anyone new is looking through the tests.

Agreed!

Reply:
Agreed. Addressed in 5b68ec5b2ec5f8b245e800857ff9dce1fb93e806. Tests should look better now.

Reply:
Yep, much more concise now. Thank you!